frontend/model/contracts/chatroom.js

```diff
@@ -223,8 +288,17 @@ sbp('chelonia/defineContract', {
         const newMessage = createMessage({ meta, hash, data: notificationData, state })
         state.messages.push(newMessage)
       },
-      sideEffect ({ contractID, hash }) {
+      sideEffect ({ contractID, hash, meta }) {
         emitMessageEvent({ contractID, hash })
+
+        if (sbp('okTurtles.data/get', 'READY_TO_JOIN_CHATROOM') || // Join by himself or Login in another device
+          sbp('okTurtles.data/get', 'JOINING_CHATROOM_ID') === contractID) { // Be added by another
+          sbp('state/vuex/commit', 'setChatRoomUnreadSince', {
+            chatRoomId: contractID,
+            messageId: hash,
+            createdDate: new Date(meta.createdDate).getTime()
```

Comment:
I am seeing a lot of repetition of this code: `new Date(meta.createdDate).getTime()`

That seems like a DRY violation. You could avoid that by passing in `meta.createdDate` directly, and then inside of `setChatRoomUnreadSince` doing the conversion.

However, why are you converting it to UNIX time anyway instead of leaving it as an ISOString?

Reply:
> However, why are you converting it to UNIX time anyway instead of leaving it as an ISOString?
That's why I wanted to use it to compare with operator `>` or `<`.